android/src/main/java/com/swmansion/reanimated/nodes/Node.java

```diff
@@ -36,7 +36,7 @@ public Node(int nodeID, @Nullable ReadableMap config, NodesManager nodesManager)
   protected abstract @Nullable Object evaluate();
 
   public final @Nullable Object value() {
-    if (mLastLoopID < mUpdateContext.updateLoopID) {
+    if (mLastLoopID < mUpdateContext.updateLoopID || mUpdateContext.contextCount > 0) {
```

Comment:
This part I'm mostly afraid of. This is tradeoff which probably is not needed. I wonder if can't we somehow bind memoization with contexts' stack. I don't want to have to decide if I want to take advantages of memoization of functional nodes. Maybe we can make a stack of memoized values as well? What do you think?

Reply:
Function nodes should be memoized already - but we need a way to force a function that is called recursively to update even though the `mLastLoogID < mUpdateContrext.updateLoopID`.

Reply:
But why can’t we memoized recursively called node in every context? I mean that probably we can make memoization context bound to evaluation context. 
I did it i my pr so it’s definitely possible!

Reply:
I tried to avoid propagating the evaluation context - we really don't have any - it is all handled by the param nodes and function call nodes. In the current implementation a function call is memoized correctly (it only overrides the `evaluate` function). What I need is a way to call a function node more than one time in the same update loop. Using a counter (`contextCount`) that is increased for each new function call we enter (and decreased when the function call is done) makes this all work - when you are in a function call, not outside function calls.